pyproject.toml

```diff
@@ -184,28 +184,71 @@ exclude_lines = [
 ]
 
 # Configure linters
-# ruff
+
+[tool.codespell]
+ignore-words-list = "emac,hist,oce,pres,vas,runn"
+
 [tool.ruff]
 fix = true
 line-length = 79
 show-fixes = true
+
 [tool.ruff.lint]
 select = ["ALL"]
+# syntax from github.com/astral-sh/ruff/issues/10878
+exclude = ["archive/*"]
 ignore = [
     # TODO: when replacing `prospector` with `ruff`, ignore rules that are
     # overly technical or conflict with other rules.
-    "E501",  # Disable line-too-long as this is taken care of by the formatter.
-    "D105",  # Disable Missing docstring in magic method as these are well defined.
+    "E501",     # Disable line-too-long as this is taken care of by the formatter.
+    "D105",     # Disable Missing docstring in magic method as these are well defined.
+    # from ESMValTool: Ignore until someone has time to fix this:
+    "B011",     # Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`
+    "UP038",    # Use `X | Y` in `isinstance` call instead of `(X, Y)`
+    # from ESMValCore: Ignore until someone has time to fix this:
+    "ANN",      # Adding type hints
+    "ARG",      # Unused argument
+    "DTZ",      # add tzinfo to datetimes
+    "ERA001",   # Found commented-out code
+    "FBT",      # avoid boolean positional arguments
+    "N818",     # Exception name should be named with an Error suffix
+    "PERF",     # Use comprehensions
+    "PGH003",   # Use specific rule codes when ignoring type issues
+    "PLC0206",  # Extracting value from dictionary without calling `.items()`
+    "PLR2004",  # Don't use magic values
+    "PLW2901",  # Don't overwrite a loop variable
+    "PT009",    # Use a regular `assert` instead of unittest-style `assertListEqual`
+    "PT011",    # pytest.raises without match argument
+    "PT027",    # Use `pytest.raises` instead of unittest-style `assertRaises`
+    "PTH",      # Use pathlib everywhere
+    "RUF022",   # `__all__` definitions that are not ordered according to an "isort-style" sort.
+    "SIM102",   # Use a single `if` statement instead of nested `if` statements
+    "SIM117",   # Use a single `with` statement with multiple contexts instead of nested `with` statements
+    "T201",     # `print` statements.
+    "TD003",    # Missing issue link on the line following this TODO
+    "TRY400",   # Uses of `logging.error` instead of `logging.exception` when logging an exception
```

Comment:
```suggestion
```
Let's not cargo cult things over from ESMValCore. We need a selection of rules for ESMValTool that warn about unreliable or unreadable code, without nitpicking (unless it can be automatically fixed).